src/libcore/cmp.rs

```diff
@@ -145,12 +150,13 @@ use self::Ordering::*;
 ///
 /// impl PartialEq<BookFormat> for Book {
 ///     fn eq(&self, other: &BookFormat) -> bool {
-///         match (&self.format, other) {
-///            (&BookFormat::Paperback, &BookFormat::Paperback) => true,
-///            (&BookFormat::Hardback,  &BookFormat::Hardback)  => true,
-///            (&BookFormat::Ebook,     &BookFormat::Ebook)     => true,
-///            (_, _) => false,
-///         }
+///         self.format == *other
+///     }
+/// }
+///
+/// impl PartialEq<Book> for BookFormat {
```

Comment:
Looks like this impl isn't needed in this example any more?

Reply:
the impetus of this pull request is that these equality implementations weren't symmetric, even though they should be: https://github.com/rust-lang/rust/issues/53844. does that make sense?

Reply:
Right, but this impl isn't being used in this example. This is the one that shows off that you can implement `PartialEq<Thing>` for multiple `Thing`s, and the assertions at the end don't use this specific formation. (You've already edited the previous example to show the asymmetric nature of PartialEq.)

Reply:
fair enough. addressed in https://github.com/rust-lang/rust/pull/57357/commits/32b28340b20ea8b45d3e663dc81bb9092ecdeeb1